File: source/frontend/src/lib/dto/response/entity/ArchitecturePrincipleResponseDto.ts
import {ArchitecturePrinciple} from "$lib/domain/entity/ar/ArchitecturePrinciple.svelte";

export class ArchitecturePrincipleResponseDto {
    id : string = "";
    title : string = "";

    static create() {
        return new ArchitecturePrincipleResponseDto();
    }
}


export function fromArchitecturalPrinciplesResponseDto(architecturalPrinciplesResponseDto : ArchitecturePrincipleResponseDto[]) : ArchitecturePrinciple[] {
    return architecturalPrinciplesResponseDto.map(fromArchitecturePrincipleResponseDto);
}

export function fromArchitecturePrincipleResponseDto(architecturePrincipleResponseDto : ArchitecturePrincipleResponseDto) : ArchitecturePrinciple {
    const architecturePrinciple : ArchitecturePrinciple = ArchitecturePrinciple.create();

    architecturePrinciple.id = architecturePrincipleResponseDto.id;
    architecturePrinciple.title = architecturePrincipleResponseDto.title;

    return architecturePrinciple;
}

File: source/backend/src/main/java/ch/unisg/backend/controller/http/dto/response/ar/ArchitecturePrincipleResponseDto.java
package ch.unisg.backend.controller.http.dto.response.ar;

import ch.unisg.backend.core.domain.entities.classes.ar.ArchitecturePrinciple;

import java.net.URI;
import java.util.HashMap;
import java.util.UUID;

public class ArchitecturePrincipleResponseDto {

    private static final String LOCATION = "http://localhost:4000/api/v1/";

    public static URI uri(UUID id) {
        return URI.create(LOCATION + id.toString());
    }

    public static HashMap<String, Object> create(ArchitecturePrinciple architecturePrinciple) {

        HashMap<String, Object> result = new HashMap<>();

        result.put("id", architecturePrinciple.getId().toString());
        result.put("title", architecturePrinciple.getTitle());

        return result;
    }
}
